src/main/java/com/amplitude/HttpTransport.java

```diff
@@ -207,10 +216,11 @@ private EventsRetryResult retryEventsOnce(String userId, String deviceId, List<E
           triggerEventCallbacks(events, response.code, response.error);
         }
         // Reduce the payload to reduce risk of throttling
-        shouldReduceEventCount = true;
+        //shouldReduceEventCount = true;
```

Comment:
why comment out?

Reply:
I found that shouldReduceEventCount was set to true for rate limit status but not payload too large status. Is this the designed behavior same on other sdks? For rate limit(throttled) situation, as I read from http api documents, events could be retry after a short period of time. Do we just drop half of the events list here without retry?

Reply:
That's the expected behavior. We have the same logic in Amplitude-Node. When it reaches the rate limit we reduce the event count for further retry. 
https://github.com/amplitude/Amplitude-Node/blob/7bb74e8d1b070cd0e5a940fac8661dde4c060e6a/packages/node/src/retry/defaultRetry.ts#L208

Reply:
I think our 'cut the half of list to drop' approach is too rough. We can retry first half and second half separately for payload too large situation or even check payload size before send out to http api.

Reply:
Added back this setting for alignment. Need to discuss the retry logic for new SDKs.